plugin.rb

```diff
@@ -123,12 +187,34 @@ def can_translate
         Jobs.enqueue(:detect_translation, post_id: object.id)
         false
       else
-        detected_lang != "DiscourseTranslator::#{SiteSetting.translator}::SUPPORTED_LANG_MAPPING".constantize[I18n.locale]
+        detected_lang != DiscourseTranslator.current_service::SUPPORTED_LANG_MAPPING[I18n.locale]
       end
     end
 
   end
 
+  add_to_serializer(:topic_list_item, :title) do
+    SiteSetting.translator_show_topic_titles_in_user_locale ? object.translated_title : object.title
```

Comment:
Having titles automatically be translated in the topic list feels a little odd because when you click through to the topic view, only the title is automatically translated, the posts aren't. 

I would also like to avoid making so many additions to the topic_list_item and topic_view serializers. Maybe we can find a way to have one button in topic lists that can trigger a translation for all the titles in the list? Not sure if that is better, but it would at least match the behaviour on the topic view.

Reply:
Hm again, I disagree here for the same reasons mentioned above. This approach was also independently suggested by both the WMF community and another multilingual community we're working with. I'd also be worried about bundling up the functionality too much here. Some communities may not want title translations.